lib/Agrammon/Web/Service.pm6

```diff
@@ -159,20 +178,39 @@ class Agrammon::Web::Service {
         my $dataset-name = %params<datasetName>;
         my $language     = %params<language>;
         my $prints       = %params<reports>;
-        my $type         = %params<type> // '';
-        my $with-filters = $type eq 'reportDetailed';
-        my $all-filters  = $type eq 'reportDetailed';
+
+        my %submission;
+        if %params<mode> and %params<mode> eq 'submission' {
+            my $sender-name = %params<senderName>;
+            # replace URL encoded newlines
+            $sender-name ~~ s:g/'%D'/\\newline\{\}/;
+            my $comment = %params<comment>;
+            $comment ~~ s:g/'%D'/\\newline\{\}/;
```

Comment:
Hm, if it's a form submission I think Cro does this automatically? Plus `%D` looks wrong, I think it's always encoded as `%` with two chars after it.

Reply:
Well, I don't understand why, but this is not working as expected. See also chat. I thus reverted to doing the manual "encoding" to XXX as I am out of ideas right now.